pkg/server/plugin/nodeattestor/k8s/sat.go

```diff
@@ -255,14 +255,15 @@ func loadServiceAccountKeys(path string) ([]crypto.PublicKey, error) {
 	for {
 		var pemBlock *pem.Block
 		pemBlock, pemBytes = pem.Decode(pemBytes)
-		if pemBlock != nil {
-			key, err := decodeKeyBlock(pemBlock)
-			if err != nil {
-				return nil, err
-			}
-			if key != nil {
-				keys = append(keys, key)
-			}
+		if pemBlock == nil {
+			return nil, satError.New("pemBlock %v, pemBytes %v", pemBlock, pemBytes)
```

Comment:
This should be the normal termination case, in other words, if pem.Decode returns no block, you should be able to `return keys, nil` here. The calling code makes sure that at least one key was loaded.

You should be able to get rid of the `if len(pemBytes) == 0` on line 268.

Reply:
Thanks for suggestions, also will fix failing tests during debugging.